doc/source/roadmap.rst

```diff
@@ -1 +1,105 @@
+is in good shape, however we can make a number of improvements:
+
+1. Library support. Our released wheels now ship with OpenBLAS, which is
+currently the only feasible performant option (ATLAS is too slow, MKL cannot be
+the default due to licensing issues, Accelerate support is dropped because
+Apple doesn't update Accelerate anymore). OpenBLAS isn't very stable though,
+sometimes its releases break things and it has issues with threading (currently
+the only issue for using SciPy with PyPy3).  We need at the very least better
+support for debugging OpenBLAS issues, and better documentation on how to build
+SciPy with it.  An option is to use BLIS for a BLAS interface (see `numpy
+gh-7372 <https://github.com/numpy/numpy/issues/7372>`__).
+
+2. Support for newer LAPACK features.  In SciPy 1.2.0 we increased the minimum
+supported version of LAPACK to 3.4.0.  Now that we dropped Python 2.7, we can
+increase that version further (MKL + Python 2.7 was the blocker for >3.4.0
+previously) and start adding support for new features in LAPACK.
+
+
+Implement sparse arrays in addition to sparse matrices
+------------------------------------------------------
+
+The sparse matrix formats are mostly feature-complete, however the main issue
+is that they act like ``numpy.matrix`` (which will be deprecated in NumPy at
+some point).  What we want is sparse *arrays* that act like ``numpy.ndarray``.
+This is being worked on in https://github.com/pydata/sparse, which is quite far
+along.  The tentative plan is:
+
+- Start depending on ``pydata/sparse`` once it's feature-complete enough (it
+  still needs a CSC/CSR equivalent) and okay performance-wise.
+- Add support for ``pydata/sparse`` to ``scipy.sparse.linalg`` (and perhaps to
+  ``scipy.sparse.csgraph`` after that).
+- Indicate in the documentation that for new code users should prefer
+  ``pydata/sparse`` over sparse matrices.
+- When NumPy deprecates ``numpy.matrix``, vendor that or maintain it as a
+  stand-alone package.
+
+
+Fourier transform enhancements
+------------------------------
+
+We want to integrate PocketFFT into ``scipy.fftpack`` for significant
+performance improvements (see `this NumPy PR
+<https://github.com/numpy/numpy/pull/11888>`__ for details),
+add a backend system to support PyFFTW and mkl-fft,
+and align the function signatures of ``numpy.fft`` and ``scipy.fftpack``.
+
+
+Support for distributed arrays and GPU arrays
+---------------------------------------------
+
+NumPy is splitting its API from its execution engine with
+``__array_function__`` and ``__array_ufunc__``.  This will enable parts of SciPy
+to accept distributed arrays (e.g. ``dask.array.Array``) and GPU arrays (e.g.
+``cupy.ndarray``) that implement the ``ndarray`` interface.  At the moment it is
+not yet clear which algorithms will work out of the box, and if there are
+significant performance gains when they do.  We want to create a map of which
+parts of the SciPy API work, and improve support over time.
```

Comment:
In addition to making use of NumPy protocols like `__array_function__`, there's the possibility of supporting these protocols in SciPy as well, e.g., so it's possible to (re)implement scipy functions like those in `scpiy.signal` for dask or gpu arrays:
http://www.numpy.org/neps/nep-0018-array-function-protocol.html#use-outside-of-numpy

Is there interest in exploring this as well?

Reply:
Yes I think so. A bit lower on the prio list probably, simply because I think we'd like to see first how this works out for numpy. But could be quite nice. What do others think?